src/resolvers/claims.ts

```diff
@@ -0,0 +1,46 @@
+import {Claim, ClaimOutcome, ClaimStatus, QueryResolvers} from "../generated/graphql";
+import {ClaimDto, ClaimOutcomeDto, ClaimStatusDto} from "../api/upstreams/claimService";
+import {LocalizedStrings} from "../translations/LocalizedStrings";
+
+const statusMap = {
+  [ClaimStatusDto.CREATED]: ClaimStatus.Submitted,
+  [ClaimStatusDto.IN_PROGRESS]: ClaimStatus.BeingHandled,
+  [ClaimStatusDto.CLOSED]: ClaimStatus.Closed,
+  [ClaimStatusDto.REOPENED]: ClaimStatus.Reopened,
+}
+
+const outcomeMap = {
+  [ClaimOutcomeDto.PAID]: ClaimOutcome.Paid,
+  [ClaimOutcomeDto.NOT_COMPENSATED]: ClaimOutcome.NotCompensated,
+  [ClaimOutcomeDto.NOT_COVERED]: ClaimOutcome.NotCompensated,
+}
+
+export const claims: QueryResolvers['claims'] = async (
+  _parent,
+  _args,
+  {upstream, strings},
+): Promise<Claim[]> => {
+  let claims : ClaimDto[] = await upstream.claimService.getMemberClaims()
+
+  claims = claims.filter(o => (!_args.status || _args.status.includes(statusMap[o.status])) && // TODO refine?
+    (!_args.outcome || _args.outcome?.includes(outcomeMap[o.outcome as ClaimOutcomeDto])));
+
+  return claims.map((c) => transformClaim(c, strings))
```

Comment:
I think you can afford making this quite a bit more verbose to highlight the filtering logic

```suggestion
  const claims = await upstream.claimService.getMemberClaims()

  const filteredClaims = claims.filter(c => {
    if (_args.status) {
      if (!_args.status.includes(statusMap[c.status])) {
        return false
      }
    }
    if (_args.outcome) {
      if (!c.outcome || !_args.outcome.includes(outcomeMap[o.outcome])) {
        return false
      }
    }
    return true
  }

  return filteredClaims.map((c) => transformClaim(c, strings))
```

Reply:
Will do!